packages/file/src/download.coffee.md

```diff
@@ -0,0 +1,363 @@
+        status = not match
+        unless status
+          @fs.remove
+            shy: true
+            target: stageDestination
+      if source_url.protocol not in protocols_http and not ssh
+        log message: "File Download without ssh (with or without cache)", level: 'DEBUG', module: 'nikita/lib/file/download'
+        hash_source = hash_target = null
+        {hash} = await @fs.hash target: config.source, algo: algo
+        hash_source = hash
+        exists = await @fs.base.exists target: config.target
+        if exists
+          {hash} = await @fs.hash target: config.target, algo: algo
+          hash_target = hash
+        match = hash_source is hash_target
+        log if match
+        then message: "Hash matches as '#{hash_source}'", level: 'INFO', module: 'nikita/lib/file/download' 
+        else message: "Hash dont match, source is '#{hash_source}' and target is '#{hash_target}'", level: 'WARN', module: 'nikita/lib/file/download'
+        status = not match
+        if status
+          @fs.mkdir
+            shy: true
+            target: path.dirname stageDestination
+          @fs.copy
+            source: config.source
+            target: stageDestination
+      if source_url.protocol not in protocols_http and ssh
+        log message: "File Download with ssh (with or without cache)", level: 'DEBUG', module: 'nikita/lib/file/download'
+        hash_source = hash_target = null
+        {hash} = await @fs.hash target: config.source, algo: algo, ssh: false, sudo: false
+        hash_source = hash
+        exists = await @fs.base.exists target: config.target
+        if exists
+          {hash} = await @fs.hash target: config.target, algo: algo
+          hash_target = hash
+        match = hash_source is hash_target
+        log if match
+        then message: "Hash matches as '#{hash_source}'", level: 'INFO', module: 'nikita/lib/file/download' 
+        else message: "Hash dont match, source is '#{hash_source}' and target is '#{hash_target}'", level: 'WARN', module: 'nikita/lib/file/download'
+        status = not match
+        if status
+          @fs.mkdir
+            shy: true
+            target: path.dirname stageDestination
+          try
+            @fs.base.createWriteStream
+              target: stageDestination
+              stream: (ws) ->
+                rs = fs.createReadStream config.source
+                rs.pipe ws
+            log message: "Downloaded local source #{JSON.stringify config.source} to remote target #{JSON.stringify stageDestination}", level: 'INFO', module: 'nikita/lib/file/download'
+          catch err
+            log message: "Downloaded local source #{JSON.stringify config.source} to remote target #{JSON.stringify stageDestination} failed", level: 'ERROR', module: 'nikita/lib/file/download'
+      log message: "Unstage downloaded file", level: 'DEBUG', module: 'nikita/lib/file/download'
+      if status
```

Comment:
I am really not fan of the status usage here, I know that javascript provide a lot/too much scope flexibility but having status modified line 320, inside an `if` block and having status used so many time makes it very confusing.